src/main/java/com/iwaconsolti/school/demo/controller/StudentsController.java

```diff
@@ -0,0 +1,71 @@
+package com.iwaconsolti.school.demo.controller;
+
+import com.iwaconsolti.school.demo.model.Grade;
+import com.iwaconsolti.school.demo.model.Student;
+import com.iwaconsolti.school.demo.service.GradeService;
+import com.iwaconsolti.school.demo.service.StudentService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/students")
+@Slf4j
+@RequiredArgsConstructor
+public class StudentsController {
+
+    private final StudentService studentService;
+    private final GradeService gradeService;
+
+    @GetMapping()
+    public ResponseEntity<List<Student>> getStudents(){
+        log.info("Getting all the students");
+        return ResponseEntity.ok(studentService.getStudents());
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Student> getStudentById(@PathVariable int id) {
+        Student student = this.studentService.getById(id);
+        log.info("Id: {}", id);
+        log.info("Student: {}", student);
+        if (student != null)
+            return ResponseEntity.ok(student);
+        return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
+    }
+
+    @PostMapping()
+    public ResponseEntity<Student> getStudents(@RequestBody Student student){
```

Comment:
The name of the method says 'getStudents' but youre creating students and sending a 'Postmapping'